carsus/io/cmfgen/cmfgen_.py

```
@@ -0,0 +1,199 @@
+        else:
+            n = None
+            line = None
+    
+    if num_row == True:
+        return n
+                    
+    return line
+
+
+to_float = lambda x: float(x.replace('D', 'E'))  # string to float, taking care of Fortran 'D' values
+
+
+class CMFGENEnergyLevelsParser(BaseParser):
+    """
+        Description
+        ----------
+        base : pandas.DataFrame
+        columns : list of str
+            (default value = COLUMNS)
+            
+        Methods
+        -------
+        load(fname)
+            Parses the input data and stores the results in the `base` attribute
+    """
+
+    def load(self, fname):
+        args = {}
+        args['header'] = None
+        args['index_col'] = False
+        args['sep'] = '\s+'
+        args['skiprows'] = find_row(fname, "Number of transitions", num_row=True)
+    
+        n = find_row(fname, "Number of energy levels").split()[0]
+        args['nrows'] = int(n)
+
+        columns = ['Configuration', 'g', 'E(cm^-1)', 'eV', 'Hz 10^15', 'Lam(A)']
+        
+        try:
+            df = pd.read_csv(fname, **args, engine='python')
+    
+        except pd.errors.EmptyDataError:
+            df = pd.DataFrame(columns=columns)
+            warnings.warn('Empty table')
+
+        # Assign column names by file content
+        if df.shape[1] == 10:
+            columns = find_row(fname, 'E(cm^-1)', "Lam").split('  ')
+            columns = list(filter(lambda x: x != '', columns))
+            columns = ['Configuration'] + columns
+            df.columns = columns
+
+        elif df.shape[1] == 7:
+            df.columns = columns + ['#']
+            df = df.drop(columns=['#'])
+
+        elif df.shape[1] == 6:
+            df.columns = range(6)  # FIXME: These files don't have column names
+
+        elif df.shape[1] == 5:
+            df.columns = columns[:-2] + ['#']
+            df = df.drop(columns=['#'])
+    
+        else:
+            warnings.warn('Inconsistent number of columns')
```

I think we should raise an Exception here.